packages/api/src/folder/controller.test.ts

```diff
@@ -0,0 +1,203 @@
+import type { Request, NextFunction } from "express";
+import { logger } from "@stela/logger";
+import request from "supertest";
+import { app } from "../app";
+import { db } from "../database";
+import {
+  extractUserEmailFromAuthToken,
+  verifyUserAuthentication,
+} from "../middleware";
+import { folderAccess } from "./access";
+
+jest.mock("../database");
+jest.mock("../middleware");
+jest.mock("@stela/logger");
+
+const setupDatabase = async (): Promise<void> => {
+  await db.sql("fixtures.create_test_accounts");
+  await db.sql("fixtures.create_test_archives");
+  await db.sql("fixtures.create_test_account_archives");
+  await db.sql("fixtures.create_test_folders");
+};
+
+const clearDatabase = async (): Promise<void> => {
+  await db.query(
+    "TRUNCATE event, account_archive, account, archive, folder CASCADE"
+  );
+};
+
+describe("patch folder", () => {
```

Comment:
I think it would be worth adding a test where the edit permission comes from the `access` table rather than `account_archive`